src/main/java/net/minecraftforge/common/crafting/CraftingHelper.java

```diff
@@ -803,13 +805,12 @@ else if (source.isDirectory())
 
                 while (itr != null && itr.hasNext())
                 {
-                    Boolean cont = processor.apply(root, itr.next());
-                    if (cont == null || !cont.booleanValue())
-                        return false;
+                    Boolean ret = processor.apply(root, itr.next());
+                    success &= ret == null ? false : ret;
```

Comment:
`ret == null ? false : ret` can be replaced with `ret != null && ret`, which is a little more consise.